### DISCUSSION TITLE
Use promise in next.config.js for createExpoWebpackConfigAsync

### DISCUSSION THREAD

**Discussion by sajadghawami** [vercel/next.js#31956] [Ideas]
### Describe the feature you'd like to request

If you want to use kitten-ui with react-native-web you currently have to [update your webpack.config like this](https://github.com/akveo/react-native-ui-kitten/issues/1535#issuecomment-944876518):

```js
const createExpoWebpackConfigAsync = require('@expo/webpack-config');

module.exports = async function(env, argv) {
    const config = await createExpoWebpackConfigAsync({
        ...env,
        babel: {
            dangerouslyAddModulePathsToTranspile: ['@ui-kitten/components']
        }
    }, argv);
    return config;
};
```

Unfortunately, nextjs has its own `next.config.js`, such as:

```js
const withPlugins = require('next-compose-plugins');
const { withExpo } = require('@expo/next-adapter');
const withTM = require('next-transpile-modules')([
  '@ui-kitten/components',
]);

module.exports = withPlugins(
  [
    [withExpo, { projectRoot: __dirname }],
    withTM({
      webpack: (config, options) => {
        config.experiments = {
          ...config.experiments,
          topLevelAwait: true,
        };
        return config;
      },
    }),
  ],
)

```

### Describe the solution you'd like

The problem here is, you [cant use a promise](https://nextjs.org/docs/messages/promise-in-next-config) which you need for `createExpoWebpackConfigAsync`.

Anyone has a fix?

## UI Kitten and Eva version

    "@eva-design/eva: "^2.1.1",
    "ui-kitten/components": "^5.1.1",
    "next": "^11.1.2",
    "next-compose-plugins": "^2.2.1",
    "next-transpile-modules": "^9.0.0",

### Describe alternatives you've considered

I ve tried all kinds of different approaches, but i think there is no way around using a promise.

**Comment by jowo-io** [comment]
looks like support was added in version `12.0.10`!

From[ the link ](https://github.com/vercel/next.js/discussions/31956#:~:text=here%20is%2C%20you-,cant%20use%20a%20promise,-which%20you%20need)you provided

> In Next.js versions above 12.0.10, module.exports = async () => is supported.

**Reply by sajadghawami** [reply]
@jowo-io oh nice, thanks for letting me know mate :)
